### PR TITLE
CPV: fix for merging IntensiveTH2F

### PR DESCRIPTION
**Pull request by sevdokim**
Hello! This is fix for proper merging of IntensiveTH2F in CPV/PhysicsTask.

### CODE REVIEW

File: Modules/CPV/include/CPV/PhysicsTask.h [hunk 1]
```diff
@@ -52,14 +52,28 @@ class IntensiveTH2F : public TH2F, public o2::mergers::MergeInterface
...
+      mCycleNumber = dynamic_cast<IntensiveTH2F*>(other)->mCycleNumber;
+      LOG(debug) << "other.fN = " << (*(dynamic_cast<IntensiveTH2F*>(other))).fN;
+      LOG(debug) << "other.binContent(1, 1) = " << (*(dynamic_cast<IntensiveTH2F*>(other))).GetBinContent(1, 1);
+      LOG(debug) << "this->fN = " << fN;
+      LOG(debug) << "this->binContent(1, 1) = " << GetBinContent(1, 1);
+      for (int iX = 1; iX <= GetNbinsX(); iX++) {
+        for (int iY = 1; iY <= GetNbinsY(); iY++) {
+          SetBinContent(iX, iY, dynamic_cast<IntensiveTH2F*>(other)->GetBinContent(iX, iY));
+        }
+      }
```
Comment (knopers8) [inline Modules/CPV/include/CPV/PhysicsTask.h:73]: Perhaps you could also do `dynamic_cast` just once at the beginning and then make sure it is not `nullptr`?
Also, will you need these debug logs for the future?

Comment (sevdokim) [reply Modules/CPV/include/CPV/PhysicsTask.h:73]: hi @knopers8 
I'm not sure how something can be `nullptr` in the merge method (asuming `this` and `other` are valid objects). But yes, you are right, it is better to use dynamic_cast once and check it for being `nullptr`.

Comment (knopers8) [reply Modules/CPV/include/CPV/PhysicsTask.h:73]: It is rather unlikely, but we've seen very unlikely events happening on grid, e.g. one of 10000 subjobs had memory issues, preventing ROOT to load correctly and generating corrupted objects. `other` could be invalid if it is `MergeInterface` but it is not `IntensiveTH2F`. It probably won't happen, but better safe than sorry.

Also, `dynamic_cast` has some overhead as compared to `reinterpret_cast`, so it's better to do it once for performance reasons.

**Comment by sevdokim** [comment]
Hi @knopers8 
I tried to implement what you proposed. However O2fst check failed without logs. Is this foreseen?

**Comment by TimoWilken** [comment]
Hi @sevdokim @knopers8, sorry, this is due to a connection issue (the error was: `Error on fetching 'EVTGEN': Error 128 from git ls-remote --heads --tags https://phab.hepforge.org/source/evtgen: fatal: unable to access 'https://phab.hepforge.org/source/evtgen/': Failed connect to phab.hepforge.org:443; Connection timed out. Aborting.`).

The check is currently re-running; it should be done soon.

**Review by Barthelemy — APPROVED** [review]

**Pull request merged**